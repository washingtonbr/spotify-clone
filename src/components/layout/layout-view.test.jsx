import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter, Link } from 'react-router-dom';
import { mount } from 'enzyme';
import LayoutView from './layout-view';

describe('LayoutView', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <LayoutView>Content</LayoutView>
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('includes link to home screen', () => {
    const wrapper = mount(
      <MemoryRouter>
        <LayoutView>Content</LayoutView>
      </MemoryRouter>,
    );

    expect(wrapper.find(Link).props().to).toBe('/');
  });
});
