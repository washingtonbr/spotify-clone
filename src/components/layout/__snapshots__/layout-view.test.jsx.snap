// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutView render corectly 1`] = `
<div
  className="sc-AxjAm xZqxm"
>
  <div
    className="sc-AxirZ eNjkev"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Spotify"
        className="sc-AxiKw gqLFEV"
        src="test-file-stub"
      />
    </a>
  </div>
  <div
    className="sc-AxhCb hNvnOw"
  >
    Content
  </div>
</div>
`;
